### PR TITLE
handles GET /api/reviews/:review_id

### PR DESCRIPTION
**Pull request by SentinelProxima**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        .then((res) => {
+            expect(res.body.categories[0].hasOwnProperty("slug")).toBe(true);
+            expect(res.body.categories[0].hasOwnProperty("description")).toBe(true);
+        });
+    });
+});
+
+describe('GET /api/reviews/:review_id', () => {
+    it('returns a review object', () => {
+        return request(app).get('/api/reviews/:1')
```
Comment (HMelesi) [inline __tests__/app.test.js]: you don't need to put the `:` in front of the id you're requesting in the path, so it should just be `/api/reviews/1`

Comment (HMelesi) [reply __tests__/app.test.js]: this applies to all of the tests on this endpoint!

File: __tests__/app.test.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        .expect(200)
+        .then((res) => {
+            expect(typeof res.body.review).toBe("object");
+        });
+    });
+    it('returns a review object with all review properties', () => {
+        return request(app).get('/api/reviews/:1')
+        .expect(200)
+        .then((res) => {
+            const rev = res.body.review;
```
Comment (HMelesi) [inline __tests__/app.test.js:41]: You could also use destructuring here - so `const { review } = res.body` but this is optional and your way is valid too.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+            expect(rev.title).toBe("Agricola");
+            expect(rev.category).toBe("euro game");
+            expect(rev.designer).toBe("Uwe Rosenberg");
+            expect(rev.owner).toBe("mallionaire");
+            expect(rev.review_body).toBe("Farmyard fun!");
+            expect(rev.review_img_url).toBe("https://www.golenbock.com/wp-content/uploads/2015/01/placeholder-user.png");
+            expect(rev.created_at).toBe("2021-01-18T10:00:20.514Z");
+            expect(rev.votes).toBe(1);
```
Comment (HMelesi) [inline __tests__/app.test.js]: You need to add the review_id to the returned review

File: __tests__/app.test.js [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+        });
+    });
+    it('returns 404 if invalid ID passed', () => {
+        return request(app).get('/api/reviews/:0')
+        .expect(404)
+        .then((res) => {
+            const msg = JSON.parse(res.text);
+            expect(msg.msg).toBe("Invalid review ID");
+        });
+    });
```
Comment (HMelesi) [inline __tests__/app.test.js]: Need to add a test in for an invalid review_id type, ie. what about a request to `/api/reviews/abc`

File: models/reviews.models.js [hunk 5]
```diff
@@ -0,0 +1,11 @@
+const db = require('../db/connection');
+
+exports.selectReview = (id_param) => {
+    const review_id = parseInt(id_param.review_id.slice(1));
```
Comment (HMelesi) [inline models/reviews.models.js]: If you don't use the `:` in the path when you make the request, you shouldn't have to do this line at all! Remove the `:` and `console.log(review_id)` here to see what comes through.

Comment (HMelesi) [reply models/reviews.models.js]: I mean remove the `:` from the request in your tests by the way, not your app.js

**Review by HMelesi — CHANGES_REQUESTED** [review]
A few things to change/add but looking great Michael! Any questions let me know

File: models/reviews.models.js [hunk 6]
```diff
@@ -0,0 +1,14 @@
+const db = require('../db/connection');
+
+exports.selectReview = (id_param) => {
+    const { review_id } = id_param;
+    if (isNaN(parseInt(review_id))) {
```
Comment (HMelesi) [inline models/reviews.models.js]: You shouldn't need to throw this error manually - PSQL will throw an error for you as the review_id is the wrong data type. You'd then be able to see the error in your error handler if you console.log(err) on line 13 of your app.js, which is where you'd send the status and message for this kind of error.

File: __tests__/app.test.js (hunk 3 above)
Comment (HMelesi) [reply __tests__/app.test.js]: This hasn't been added yet that I can see?

**Review by HMelesi — CHANGES_REQUESTED** [review]
Nearly there Michael, just a change on how to handle that invalid review_id data type using the PSQL error code and our error handlers.

File: __tests__/app.test.js (hunk 3 above)
Comment (SentinelProxima) [reply __tests__/app.test.js:49]: Sorry, I don't know what this means

**Review by xViolaine — APPROVED** [review]

**Pull request merged**